### PR TITLE
Prevents having classes/groups/exams with the same name.

### PR DESCRIPTION
**Pull request by christianmemije**
## Summary

* Prevents having classes/groups/exams with the same name.
* Fixes https://github.com/learningequality/kolibri/issues/1279
* Fixes https://github.com/learningequality/kolibri/issues/1224
* Deletes a lot of useless code.

peek 2017-04-24 23-10

### CODE REVIEW

File: kolibri/auth/serializers.py [hunk 1]
```diff
@@ -79,6 +79,11 @@ class ClassroomSerializer(serializers.ModelSerializer):
     coach_count = serializers.SerializerMethodField()
     admin_count = serializers.SerializerMethodField()
 
+    def validate_name(self, value):
```
Comment (rtibbles) [inline kolibri/auth/serializers.py:82]: One alternative here would be to use the UniqueTogetherValidator on `name` and `parent`:
http://www.django-rest-framework.org/api-guide/validators/#uniquetogethervalidator

File: kolibri/auth/serializers.py [hunk 2]
```diff
@@ -97,6 +102,11 @@ class LearnerGroupSerializer(serializers.ModelSerializer):
 
     user_ids = serializers.SerializerMethodField()
 
+    def validate_name(self, value):
+        if LearnerGroup.objects.filter(name__iexact=value, parent=self.parent).exists():
```
Comment (rtibbles) [inline kolibri/auth/serializers.py:106]: As above.

File: kolibri/content/serializers.py [hunk 3]
```diff
@@ -177,6 +178,11 @@ class Meta:
         )
         read_only_fields = ('creator',)
 
+    def validate_title(self, value):
+        if Exam.objects.filter(title__iexact=value, collection=self.initial_data['collection']).exists():
```
Comment (rtibbles) [inline kolibri/content/serializers.py:182]: Can do UniqueTogether on `title` and `collection`.

**Comment by indirectlylit** [comment]
addressed validators in f0d8e9c3215b883c0865338c55468b3e8652dad5

**Pull request merged**